### PR TITLE
[MM-58443] Migrate tooltips of 'components/global_header/right_controls/saved_posts_button/saved_posts_button.tsx' to WithTooltip

### PR DESCRIPTION
**Pull request by suraj-anthwal**
#### Summary

Eliminate the usage of Overlay Trigger and Tooltip component.
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27178
#### Screenshots

No screenshots
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
Hi @suraj-anthwal thank you for your contribution can you check CLA as per https://github.com/mattermost/mattermost/pull/27185#issuecomment-2135912195

File: webapp/channels/src/components/global_header/right_controls/saved_posts_button/saved_posts_button.tsx [hunk 1]
```diff
@@ -31,34 +29,31 @@ const SavedPostsButton = (): JSX.Element | null => {
...
+            size={'sm'}
+            icon={'bookmark-outline'}
+            toggled={rhsState === RHSStates.FLAG}
+            onClick={savedPostsButtonClick}
+            inverted={true}
+            compact={true}
+            aria-expanded={rhsState === RHSStates.FLAG}
+            aria-controls='searchContainer' // Must be changed if the ID of the container changes
+            aria-label={formatMessage({id: 'channel_header.flagged', defaultMessage: 'Saved messages'})}
+        />
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/global_header/right_controls/saved_posts_button/saved_posts_button.tsx:43]: please move this under with WithTooltip.

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/global_header/right_controls/saved_posts_button/saved_posts_button.tsx (hunk 1 above)
Comment (suraj-anthwal) [reply webapp/channels/src/components/global_header/right_controls/saved_posts_button/saved_posts_button.tsx]: Hey @M-ZubairAhmed , addressed the above!

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Can you please check failing ci checks please

**Comment by M-ZubairAhmed** [comment]
@suraj-anthwal do you have any progress on this one?

**Comment by suraj-anthwal** [comment]
@M-ZubairAhmed , I have not reworked on this one.

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed , I have not reworked on this one.

Do you plan on completing this one?

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by Rajat-Dabade — APPROVED** [review]

**Pull request merged**